common/src/main/scala/org/genivi/sota/http/NamespaceDirectives.scala

```diff
@@ -10,16 +12,28 @@ import org.slf4j.LoggerFactory
 import scala.util.Try
 
 object NamespaceDirectives {
+  import akka.http.scaladsl.server.Directives._
+
   lazy val logger = LoggerFactory.getLogger(this.getClass)
 
   import eu.timepit.refined.refineV
 
+  val NAMESPACE = "x-ats-namespace"
+
+  def nsHeader(ns: Namespace): HttpHeader = RawHeader(NAMESPACE, ns.get)
+
   def configNamespace(config: Config): Namespace = {
     Namespace( Try(config.getString("core.defaultNs")).getOrElse("default"))
   }
 
-  lazy val defaultNamespaceExtractor: Directive1[Namespace] =
-    BasicDirectives.provide(configNamespace(ConfigFactory.load()))
+  lazy val defaultNamespaceExtractor: Directive1[Namespace] = {
+    extractRequest flatMap { req =>
```

Comment:
we should prefer `.flatMap` http://docs.scala-lang.org/style/method-invocation.html